File: cpp/program/Produk.cpp
#include <string>
using namespace std;

class Produk {
protected:
    string idProduk;
    string nama;
    string deskripsi;
    double harga;

public:
    Produk(string id = "", string n = "", string d = "", double h = 0)
        : idProduk(id), nama(n), deskripsi(d), harga(h) {}

    string getIdProduk() const { return idProduk; }
    string getNama() const { return nama; }
    string getDeskripsi() const { return deskripsi; }
    double getHarga() const { return harga; }
};

File: java/program/Produk.java
public class Produk {
    protected String idProduk;
    protected String nama;
    protected String deskripsi;
    protected double harga;

    public Produk(String idProduk, String nama, String deskripsi, double harga) {
        this.idProduk = idProduk;
        this.nama = nama;
        this.deskripsi = deskripsi;
        this.harga = harga;
    }

    public String getIdProduk() { return idProduk; }
    public String getNama() { return nama; }
    public String getDeskripsi() { return deskripsi; }
    public double getHarga() { return harga; }
}
